electronics/apparent_power.py

```diff
@@ -1,25 +1,34 @@
+import cmath
 import math
 
 
-def apparent_power(real_power: float, reactive_power: float) -> float:
+def apparent_power(voltage: float, current: float, voltage_angle: float, current_angle: float) -> complex:
     """
-    Calculate apparent power from real power and reactive power.
-
-    Examples:-
-    >>> apparent_power(100, 50)
-    111.80339887498948
-    >>> apparent_power(100, -50)
-    111.80339887498948
-    >>> apparent_power(-100, 50)
-    111.80339887498948
-    >>> apparent_power(-100, -50)
-    111.80339887498948
-    >>> apparent_power(0, 0)
-    0.0
+    Calculate the apparent power in a single-phase AC circuit.
+        
+    >>> apparent_power(100, 5, 0, 0)
+    (500+0j)
+    >>> apparent_power(100, 5, 90, 0)
+    (3.061616997868383e-14+500j)
+    >>> apparent_power(100, 5, -45, -60)
+    (-129.40952255126032-482.9629131445341j)
+    >>> apparent_power(200, 10, -30, -90)
+    (-999.9999999999998-1732.0508075688776j)
     """
-    return math.sqrt(real_power**2 + reactive_power**2)
+    # Convert angles from degrees to radians
+    voltage_angle_rad = math.radians(voltage_angle)
+    current_angle_rad = math.radians(current_angle)
+
+    # Convert voltage and current to rectangular form
+    voltage_rect = cmath.rect(voltage, voltage_angle_rad)
+    current_rect = cmath.rect(current, current_angle_rad)
+
+    # Calculate apparent power
+    apparent_power = voltage_rect * current_rect
 
+    return apparent_power
 
+# Doctests
```

Comment:
This comment just repeats what the code just below it says so it can be dropped.